File: trunk/src/cpp/opinion.cpp
#include "engines/log.hpp"
#include "opinion.hpp"
#include "world.hpp"

using namespace std;
using namespace Radakan;

//	static
string Opinion ::
	get_class_name ()
{
	return "Opinion";
}

//  constructor
Opinion ::
	Opinion (const Reference <Object> & other) :
	Object ("opinion about " + other . get_name ()),
	about (other . get_name (true)),
	like (0.5),
	fear (0)
{
	Engines :: Log :: trace (me, Opinion :: get_class_name (), "", other . get_name ());

	assert (Opinion :: is_initialized ());
}

//  destructor
Opinion ::
	~Opinion ()
{
	Engines :: Log :: trace (me, Opinion :: get_class_name (), "~");
	assert (Opinion :: is_initialized ());

	prepare_for_destruction ();
}

//	virtual
bool Opinion ::
	is_initialized ()
	const
{
	assert (Object :: is_initialized ());

	return true;
}

File: trunk/radakan/src/com/gibbon/radakan/ai/Opinion.java
package com.gibbon.radakan.ai;

/**
 *
 * @author t
 */
public final class Opinion {
    public Opinion(Character new_about) {
        about = new_about;
    }

    final public Character about;

    public Float like = new Float(0.5);
    public Float fear = new Float(0);
}


